app/payments/razorpay.php

```diff
@@ -17,172 +21,68 @@
-                        if($http_status === 200 and isset($response_array['error']) === false){
-                            $success = true;    
-                        }
-                        else {
-                            $success = false;
-
-                            if(!empty($response_array['error']['code'])) {
-                                $error = $response_array['error']['code'].":".$response_array['error']['description'];
-                            }
-                            else {
-                                $error = "RAZORPAY_ERROR:Invalid Response <br/>".$result;
-                            }
-                        }
-                    }
-                        
-                    //close connection
-                    curl_close($ch);
-                }
-                catch (Exception $e) {
-                    $success = false;
-                    $error ="CSCART_ERROR:Request to Razorpay Failed";
-                }
-
-                if($success === true){
-                    $pp_response['order_status'] = 'P';
-                    $pp_response['reason_text'] = fn_get_lang_var('text_rzp_success');
-                    $pp_response['transaction_id'] = @$order;
-                    $pp_response['client_id'] = $razorpay_payment_id;
-
-                    fn_finish_payment($merchant_order_id, $pp_response);
-                    fn_order_placement_routines('route', $merchant_order_id);
-                }
-                else {
-                    $pp_response['order_status'] = 'O';
-                    $pp_response['reason_text'] = fn_get_lang_var('text_rzp_pending').$error;
-                    $pp_response['transaction_id'] = @$order;
-                    $pp_response['client_id'] = $razorpay_payment_id;
-
-                    fn_finish_payment($merchant_order_id, $pp_response);
-                    fn_set_notification('E', __('error'), __('text_rzp_failed_order').$merchant_order_id);
-                    fn_order_placement_routines('checkout_redirect');
-                }
-
-            }
-        }
-        else {
-            fn_set_notification('E', __('error'), __('text_rzp_failed_order').$_REQUEST['merchant_order_id']);
-            fn_order_placement_routines('checkout_redirect');
-        }
+        $razorpayPayment = new RazorpayPayment();
+
+        $razorpayPayment->processRazorpayResponse();
+        exit;
     }
-    exit;
 }
-else {
+else
+{
+    $razorpayPayment = new RazorpayPayment();
+
     $url = fn_url("payment_notification.return?payment=razorpay", AREA, 'current');
-    $checkout_url = "https://checkout.razorpay.com/v1/checkout.js";
+
+    $data = $razorpayPayment->getOrderData($order_id, $order_info, $processor_data);
+
+    $keyId = $processor_data['processor_params']['key_id'];
+
+    $keySecret = $processor_data['processor_params']['key_secret'];
+
+    $api = new Api($keyId, $keySecret);
+
+    $razorpayOrder = $api->order->create($data);
```

Comment:
Add a try catch around this line and echo the exception. 

echo 'CS Cart Error : ' . $e->getMessage();